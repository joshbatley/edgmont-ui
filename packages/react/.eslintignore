**/lib/*
**/node_modules/
**/jest.config.js
**/setupTests.ts
**/rollup.config.js
**/dist/*
**/src/tailwind.css
**/docs/*
**/preset.js
