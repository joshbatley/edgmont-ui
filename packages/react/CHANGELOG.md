# @edgmont-ui/react

## 1.0.0

### Major Changes

- b4b59a8: Revamped all components, add themes and fix broken logix

## 0.3.0

### Minor Changes

- cc0ffd7: Updated readme and version

## 3.0.1

### Patch Changes

- 1c12ec6: readme

## 3.0.0

### Major Changes

- 52e9e42: asdas

## 2.0.2

### Patch Changes

- 7e13337: asd

## 2.0.1

### Patch Changes

- cba70fc: asdada

## 2.0.0

### Major Changes

- 323e843: ad

### Minor Changes

- a80937b: asd

## 1.0.0

### Major Changes

- c4b8b3f: asd
