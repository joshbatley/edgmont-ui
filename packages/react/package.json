{
  "name": "@edgmont-ui/react",
  "version": "1.0.2",
  "private": false,
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "homepage": "https://ui.edgmont.dev/",
  "bugs": {
    "url": "https://github.com/joshbatley/edgmont-ui/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/joshbatley/edgmont-ui"
  },
  "jest": {
    "preset": "@repo/jest-presets/jest/browser"
  },
  "contributors": [
    "joshbatley"
  ],
  "files": [
    "dist",
    "package.json"
  ],
  "scripts": {
    "build": "tsup src/index.ts --format esm,cjs --dts --external react",
    "dev": "tsup src/index.ts --format esm,cjs --watch --dts --external react",
    "lint": "eslint . --max-warnings 0",
    "test": "jest",
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist"
  },
  "devDependencies": {
    "@repo/eslint-config": "workspace:*",
    "@repo/jest-presets": "workspace:*",
    "@repo/typescript-config": "workspace:*",
    "@turbo/gen": "^1.13.3",
    "@types/eslint": "^8.56.10",
    "@types/node": "^20.12.12",
    "@types/react": "^18.3.2",
    "@types/react-dom": "^18.3.0",
    "eslint": "^8.57.0",
    "react": "^18.3.1",
    "typescript": "5.4.5"
  },
  "dependencies": {
    "@heroicons/react": "^2.1.3",
    "@popperjs/core": "^2.11.8",
    "@testing-library/jest-dom": "^6.4.5",
    "@testing-library/react": "^14.3.1",
    "@testing-library/react-hooks": "^8.0.1",
    "@testing-library/user-event": "^14.5.2",
    "@types/jest": "^29.5.12",
    "@types/react-datepicker": "^6.2.0",
    "@types/styled-components": "^5.1.34",
    "@types/styled-system": "^5.1.22",
    "date-fns": "^3.6.0",
    "downshift": "^8.5.0",
    "jest": "^29.7.0",
    "jest-styled-components": "^7.2.0",
    "polished": "^4.3.1",
    "prism-react-renderer": "^2.3.1",
    "react-datepicker": "^6.9.0",
    "react-dom": "^18.3.1",
    "react-dropzone": "^14.2.3",
    "react-focus-lock": "^2.12.1",
    "react-popper": "^2.3.0",
    "react-simple-code-editor": "^0.13.1",
    "react-spring": "^9.7.3",
    "react-use": "^17.5.0",
    "styled-components": "^5.3.11",
    "styled-system": "^5.1.5",
    "tsup": "^8.0.2"
  }
}