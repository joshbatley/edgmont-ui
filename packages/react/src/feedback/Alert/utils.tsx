import React from 'react';
import {
  ExclamationCircleIcon,
  ExclamationTriangleIcon,
  InformationCircleIcon,
  CheckCircleIcon,
} from '@heroicons/react/24/solid';

export const getIcon = (severity: Severity) => {
  let props = {
    height: 22,
    width: 22,
  };
  switch (severity) {
    case 'error':
      return <ExclamationCircleIcon {...props} />;
    case 'warning':
      return <ExclamationTriangleIcon {...props} />;
    case 'info':
      return <InformationCircleIcon {...props} />;
    case 'success':
      return <CheckCircleIcon {...props} />;
  }
};

