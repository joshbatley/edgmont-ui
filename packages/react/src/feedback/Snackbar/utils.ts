import { IntercardinalPoints } from '../../types';

export const createId = () => {
  return Math.random().toString(36).substr(2, 9);
};

export const getPosition = (position: IntercardinalPoints) => {
  switch (position) {
    case 'NE':
      return `
      top: 0;
      right: 0;
      flex-direction: column;
      `;
    case 'NW':
      return `
      top: 0;
      left: 0;
      flex-direction: column;
      `;
    case 'SE':
      return `
      right: 0;
      bottom: 0;
      flex-direction: column-reverse;
      `;
    case 'SW':
      return `
        left: 0;
        bottom: 0;
        flex-direction: column-reverse;
      `;
  }
};
