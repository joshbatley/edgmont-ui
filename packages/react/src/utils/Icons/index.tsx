
export const RadioIcon = ({ ...rest }) => (
  <svg viewBox='0 0 16 16' fill='currentColor' xmlns='http://www.w3.org/2000/svg' {...rest}>
    <circle cx='8' cy='8' r='3' />
  </svg>
);
