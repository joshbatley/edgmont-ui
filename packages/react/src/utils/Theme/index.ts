export * from './Theme';
export * from './DefaultThemeProvider';
export * from './GlobalStyles';
