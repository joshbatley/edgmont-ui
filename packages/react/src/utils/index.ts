export * from './Backdrop';
export * from './ClickOutside';
export * from './FocusLock';
export * from './Portal';
export * from './Scrollable';
export * from './Transitions';
export * from './Icons';
export * from './Theme';
