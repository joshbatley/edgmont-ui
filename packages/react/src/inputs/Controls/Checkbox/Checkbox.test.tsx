import { render as testRender } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { EdgmontUI } from '../../../utils';
import { Checkbox } from '.';

function render(Comp: any) {
  return testRender(<EdgmontUI>{Comp}</EdgmontUI>);
}

const onChangeMock = jest.fn();

describe('Checkbox test', () => {
  it('should render', () => {
    const { container } = render(<Checkbox />);
    expect(container).not.toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
  it('should render with disabled classes', () => {
    const { container } = render(<Checkbox disabled />);
    expect(container).not.toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });

  it('should update value when updated', async () => {
    const { getByRole } = render(<Checkbox value="testing-value" name="testing-input" onChange={onChangeMock} />);
    expect(onChangeMock).not.toHaveBeenCalled();
    await userEvent.click(getByRole('checkbox', { hidden: true }));
    expect(onChangeMock).toHaveBeenCalledTimes(1);
  });
});
