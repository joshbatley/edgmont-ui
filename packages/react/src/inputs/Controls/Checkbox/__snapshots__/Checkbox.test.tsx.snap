// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox test should errored with disabled classes 1`] = `
<div>
  <label
    class="cursor-pointer group flex space-x-2 items-center"
  >
    <div
      class="relative h-4 w-4 border bg-white overflow-hidden shadow-sm flex items-center justify-center mr-2 rounded-md border-red-300 shadow-red-200"
    >
      <input
        class="peer hidden"
        type="checkbox"
      />
      <div
        class="bg-white w-full h-full absolute peer-checked:bg-primary-500 group-hover:peer-checked:bg-primary-600 peer-checked:bg-red-500 group-hover:peer-checked:bg-red-600"
      />
      <svg
        aria-hidden="true"
        class="group-hover:text-primary-500 text-transparent z-10 relative peer-checked:text-white"
        fill="none"
        height="16"
        stroke="currentColor"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 13l4 4L19 7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </label>
</div>
`;

exports[`Checkbox test should render 1`] = `
<div>
  <label
    class="cursor-pointer group flex space-x-2 items-center"
  >
    <div
      class="relative h-4 w-4 border bg-white overflow-hidden shadow-sm flex items-center justify-center mr-2 rounded-md"
    >
      <input
        class="peer hidden"
        type="checkbox"
      />
      <div
        class="bg-white w-full h-full absolute peer-checked:bg-primary-500 group-hover:peer-checked:bg-primary-600"
      />
      <svg
        aria-hidden="true"
        class="group-hover:text-primary-500 text-transparent z-10 relative peer-checked:text-white"
        fill="none"
        height="16"
        stroke="currentColor"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 13l4 4L19 7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </label>
</div>
`;

exports[`Checkbox test should render with disabled classes 1`] = `
<div>
  <label
    class="cursor-not-allowed group flex space-x-2 items-center"
  >
    <div
      class="relative h-4 w-4 border bg-white overflow-hidden shadow-sm flex items-center justify-center mr-2 rounded-md"
    >
      <input
        class="peer hidden"
        disabled=""
        type="checkbox"
      />
      <div
        class="bg-white w-full h-full absolute bg-gray-300 peer-checked:bg-gray-400"
      />
      <svg
        aria-hidden="true"
        class="text-transparent z-10 relative peer-checked:text-white"
        fill="none"
        height="16"
        stroke="currentColor"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 13l4 4L19 7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </label>
</div>
`;
